openadapt/app/tray.py

```diff
@@ -185,6 +226,123 @@ def _quit() -> None:
+        """Shows unzip started toast."""
+        self.show_toast(message)
+        self.cancel_download_event.clear()
+
+    def download_latest_version(self, base_url: str, latest_version: str) -> None:
+        """Download latest version of the app."""
+        if sys.platform == "darwin":
+            file_ext = ".app.zip"
+        else:
+            file_ext = ".zip"
+
+        file_name = f"OpenAdapt-v{latest_version}{file_ext}"
+        download_url = base_url + f"/v{latest_version}/{file_name}"
+
+        downloads_path = os.path.join(os.path.expanduser("~"), "Downloads")
+        local_filename = os.path.join(downloads_path, file_name)
+
+        response = requests.get(download_url, stream=True)
+        total_size = response.headers.get("content-length")
+        total_size = int(total_size) if total_size else None
+        block_size = 1024  # 1 Kilobyte
+        start_time = time.time()
+        last_update_time = start_time
+        try:
+            with open(local_filename, "wb") as file, tqdm(
+                total=total_size, unit="B", unit_scale=True, desc=file_name
+            ) as progress_bar:
+                for data in response.iter_content(block_size):
+                    if self.cancel_download_event.is_set():
+                        self.cancel_download_event.clear()
+                        return
+                    file.write(data)
+                    progress_bar.update(len(data))
+                    current_time = time.time()
+
+                    if (
+                        current_time - last_update_time > DOWNLOAD_TOAST_UPDATE_TIME
+                        and progress_bar.n > 0
+                        and progress_bar.n < progress_bar.total
+                    ):
+                        elapsed_time = current_time - start_time
+                        last_update_time = current_time
+                        rate = progress_bar.n / elapsed_time
+                        eta_seconds_actual = (
+                            progress_bar.total - progress_bar.n
+                        ) / rate
+                        # convert to minutes
+                        eta_minutes = int(eta_seconds_actual // 60)
+                        eta_seconds_formatted = eta_seconds_actual % 60
+                        eta_formatted = (
+                            f"{eta_minutes}:{int(eta_seconds_formatted):02d}"
+                        )
+
+                        progress_message = (
+                            f"Estimated time remaining: {eta_formatted} minutes"
+                        )
+                        self.download_start_toast.emit(progress_message)
+
+            self.unzipping_started_toast.emit("Unzipping Started")
+            unzip_file(local_filename)
+            self.download_complete.emit("Download & Unzipping Complete")
+        except Exception as e:
+            logger.error(e)
+            self.download_complete.emit("Download Failed")
```

Comment:
I would expect this message to be symmetrical to the success case, e.g.:

```
"Download & Unzipping Complete"
...
"Download & Unzipping Failed"
```

Perhaps instead of `Download & Unzipping` it can be simply `Update`

Also, should we rename `download_complete` to e.g. `update_complete`?